### PR TITLE
Add `quick-fork-deletion` feature

### LINKED ISSUE #3730: Ease repository deletion
I hate deleting repositories because at least 5 clicks + selections, copy-paste, scroll + password. 

I think this can be simplified, at the very least for forks. For example, if there are:

- no stars
- no open issues
- no local PRs
- no cross-fork PRs (via `show-open-prs-of-forks`)

We could show a `Delete fork` tab next to the Settings tab, or in `more-dropdown`

Such link would:

1. show a full-page 5-second deletion countdown  ("click anywhere to stop")
3. call the API
4. load github.com
	- or, if the tab is in the background by then and if possible, we could close the tab once it's done (via `browser.tabs.remove()`)

### PR DESCRIPTION
**Pull request by fregante**
Fixes #3730 

v4

[v3](https://user-images.githubusercontent.com/1402241/99633143-0a8a3e80-2a04-11eb-89df-2c24b2d0299c.gif), [v2](https://user-images.githubusercontent.com/1402241/99606459-71d9cb80-29cf-11eb-9ec0-9b7ec498ee77.gif), [v1](https://user-images.githubusercontent.com/1402241/99603046-2a037600-29c8-11eb-8da2-cc412ba17c4c.gif)

## Next

- [x] Ensure that the token has the `delete_repo` scope first: https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps
    ✅ Done: [gif](https://user-images.githubusercontent.com/1402241/99718841-237d0900-2a71-11eb-97d3-9814ac91805b.gif)

- [x] If user is not the owner, ensure that can admin the current repo: https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-repository-permissions-for-a-user
- [x] Only show when there are no local issues, PRs, stars
- [x] Cleanup

## Test

The feature depends on the current user, so do the testing URLs (these links only apply to me):

- Fork you don't own (no button): https://github.com/0x01x0/refined-github
- Fork you own, with at least 1 star (no button)
- Fork you own, with issues or PRs (button but will `confirm()`): https://github.com/fregante/gatsby
- Fork you own, without issues/PRs (button without confirmation): https://github.com/fregante/refined-github

### CODE REVIEW

File: source/features/quick-fork-deletion.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+async function handleFirstClick(event: delegate.Event): Promise<void> {
+	const button = event.delegateTarget;
+	button.classList.remove('btn-sm');
+	event.preventDefault();
+
+	let secondsLeft = 5;
+	do {
+		button.textContent = `Deleting fork in ${pluralize(secondsLeft, '$$ second')}`;
+		await delay(1000); // eslint-disable-line no-await-in-loop
+	} while (--secondsLeft);
```
Comment (yakov116) [inline source/features/quick-fork-deletion.tsx:30]: ❤️

Comment (fregante) [reply source/features/quick-fork-deletion.tsx:30]: Not sure about all of this, I just wanted to create a PR 🤣

Comment (fregante) [reply source/features/quick-fork-deletion.tsx:41]: I hate this pattern, but can't think of anything better (recursion would be nice I guess...)

Comment (fregante) [reply source/features/quick-fork-deletion.tsx:40]: Yikes, no:

```js
	return (async function down(secondsLeft: number): Promise<boolean> {
		if (abortController.signal.aborted) {
			return false;
		}

		if (!secondsLeft) {
			return true;
		}

		button.textContent = `Deleting fork in ${pluralize(secondsLeft, '$$ second')}. Cancel?`;
		await delay(1000);
		return down(secondsLeft - 1);
	})(5);
```

File: source/features/quick-fork-deletion.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		[data-hotkey="g i"] .Counter:not([hidden]),
+		[data-hotkey="g p"] .Counter:not([hidden]),
+		.rgh-open-prs-of-forks
+	`);
+
+	if (hasContent && !confirm('This fork has open issues/PRs, are you sure you want to delete everything?')) {
+		// Close the <details> element again
+		event.delegateTarget.open = false;
+	} else {
+		setTimeout(start, 1, event.delegateTarget); // Without the timeout, the same toggle event will also trigger the AbortController
```
Comment (fregante) [inline source/features/quick-fork-deletion.tsx:28]:
```suggestion
		// Without the timeout, the same toggle event will also trigger the AbortController
		setTimeout(start, 1, event.delegateTarget); 
```

File: source/features/quick-fork-deletion.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import pluralize from '../helpers/pluralize';
+import addNotice from '../github-widgets/notice-bar';
+import looseParseInt from '../helpers/loose-parse-int';
+
+function handleToggle(event: delegate.Event<Event, HTMLDetailsElement>): void {
+	const hasContent = select.exists(`
+		[data-hotkey="g i"] .Counter:not([hidden]),
+		[data-hotkey="g p"] .Counter:not([hidden]),
+		.rgh-open-prs-of-forks
+	`);
```
Comment (fregante) [inline source/features/quick-fork-deletion.tsx:22]:
```suggestion
	const hasContent = select.exists([
		'[data-hotkey="g i"] .Counter:not([hidden])', // Open issues
		'[data-hotkey="g p"] .Counter:not([hidden])', // Open PRs
		'.rgh-open-prs-of-forks' PRs opened in the source repo
	]);
```

**Pull request merged**